src/GithubMergeTool/GithubMergeTool.cs

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+
+namespace GithubMergeTool
+{
+    public class GithubMergeTool
+    {
+        private static readonly Uri GithubBaseUri = new Uri("https://api.github.com/");
+
+        private readonly HttpClient _client;
+
+        public GithubMergeTool(
+            string username,
+            string password)
+        {
+            var client = new HttpClient();
+            client.BaseAddress = GithubBaseUri;
+
+            var authArray = Encoding.ASCII.GetBytes($"{username}:{password}");
+            client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue(
+                "Basic",
+                Convert.ToBase64String(authArray));
+            client.DefaultRequestHeaders.Add(
+                "user-agent",
+                "Mozilla/4.0 (compatible; MSIE 6.0; Windows NT 5.2;)");
+
+            _client = client;
+        }
+
+        /// <summary>
+        /// Create a merge PR.
+        /// </summary>
+        /// <returns>
+        /// (true, null) if the PR was created without error.
+        /// (false, null) if the PR wasn't created due to a PR already existing
+        /// or if the <paramref name="destBranch"/> contains all the commits
+        /// from <paramref name="srcBranch"/>.
+        /// (false, error response) if there was an error creating the PR.
+        /// </returns>
+        public async Task<(bool prCreated, HttpResponseMessage error)> CreateMergePr(
+            string repoOwner,
+            string repoName,
+            string srcBranch,
+            string destBranch)
+        {
+            string prTitle = $"Merge {srcBranch} to {destBranch}";
+
+            // Check to see if there's already a PR
+            // TODO: handle pagination
```

Comment:
We're too good to use Octokit? :-)

Reply:
Dependencies in Azure functions are tricky. I'd rather not add more.